independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateImpl.java

```diff
@@ -79,6 +80,17 @@ public String render() {
                     }));
         }
 
+        @Override
+        public Uni<String> createUni() {
+            return Uni.createFrom().emitter(emitter -> renderAsync().whenComplete((r, f) -> {
```

Comment:
When not just `Uni.createFrom().completionStage(() -> renderAsync())` ?

Reply:
Ah, I missed the `UniCreate.completionStage(Supplier<? extends CompletionStage<? extends T>>)` variant. All the reactive APIs are so rich ;-).